### PR TITLE
Same connection and GPS checks in all examples

### PR DESCRIPTION
**Pull request by potaito**
Context: https://github.com/mavlink/MAVSDK-Python/issues/369#issuecomment-1073729797

A year ago I extended the check prior to arming from 
```
if health.is_global_position_ok
```
to 
```
if health.is_global_position_ok and health.is_home_position_ok:
```
because otherwise the PX4 preflight checks might fail if the python scripts are too quickly attempting to arm the drone after it acquired GPS.

Unfortunately I added this patch to only a few examples and not all of them. This PR is making up for that, and it also unifies the connection check at the beginning of almost every script. I don't mind the exact phrasing of the messages and whether it's "drone discovered" or "drone connected", but I think it's good to have it similar in all example scripts.

### CODE REVIEW

**Comment by potaito** [comment]
@JonasVautherin please take a look :)

File: examples/failure_injection.py [hunk 1]
```diff
@@ -12,13 +12,13 @@ async def run():
     print("Waiting for drone to connect...")
     async for state in drone.core.connection_state():
         if state.is_connected:
-            print(f"Drone discovered with UUID: {state.uuid}")
+            print(f"-- Connected to drone!")
```
Comment (potaito) [inline examples/failure_injection.py:15]: If I recall correctly, `{state.uuid}` was causing issues with some MAVSDK versions. At least I got errors sometimes trying to run this script. Should we just remove it or is there a new "proper way" to access the UUID?

Comment (JonasVautherin) [reply examples/failure_injection.py:15]: Now the UUID is part of the info plugin, I think :+1:

**Review by JonasVautherin — APPROVED** [review]
Thanks!

**Pull request merged**